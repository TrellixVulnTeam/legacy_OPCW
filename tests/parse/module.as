# Declare a module.
module math {

    # Declare a type in this module.
    struct Point { x: int, y: int }

    # Declare a few variables in this module.
    static a: int = 23;
    let b: float64 = 42.2;
    let mut o: Point;

    # Declare a function in this module.
    def pow(a: float64, mut i: uint): float64 { }

}

def main() {

    # Declare a variable of a type from the module.
    let o: math.Point;

    # Invoke a function from the module.
    let result: float64 = math.pow(math.b, 2);

}
