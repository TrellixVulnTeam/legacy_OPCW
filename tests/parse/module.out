ModuleDecl <?> _
  Nodes <?> 
    ModuleDecl <?> math
      Nodes <?> 
        Struct <?> Point
          Type Parameters <?> 
          Members <?> 
            StructMem <?> x
              Ident <?> int
            StructMem <?> y
              Ident <?> int
        SlotDecl <?> a
          Ident <?> int
          IntegerExpr <?> 23 (10)
        SlotDecl <?> b
          Ident <?> float64
          FloatExpr <?> 42.2
        SlotDecl <?> mut o
          Ident <?> Point
        FuncDecl <?> pow
          Ident <?> float64
          Type Parameters <?> 
          Parameters <?> 
            FuncParam <?> a
              Ident <?> float64
            Mutable FuncParam <?> mut i
              Ident <?> uint
          Block <?>
            Nodes <?> 
    FuncDecl <?> main
      Type Parameters <?> 
      Parameters <?> 
      Block <?>
        Nodes <?> 
          SlotDecl <?> o
            MemberExpr <?>
              Ident <?> math
              Ident <?> Point
          SlotDecl <?> result
            Ident <?> float64
            CallExpr <?>
              MemberExpr <?>
                Ident <?> math
                Ident <?> pow
              Arguments <?> 
                Argument <?>
                  MemberExpr <?>
                    Ident <?> math
                    Ident <?> b
                Argument <?>
                  IntegerExpr <?> 2 (10)
